src/App.js

```diff
@@ -1,30 +1,60 @@
-import { useEffect, useState } from "react"
-import { Link, Route, Routes } from "react-router-dom"
-import ContactsList from "./components/ContactsList"
-import ContactsAdd from "./components/ContactsAdd"
-import ContactsView from "./components/ContactsView"
-import "./styles/styles.css"
+import { useEffect, useState } from "react";
+import { Link, Routes, Route } from "react-router-dom";
+import ContactsList from "./components/ContactsList";
+import ContactsAdd from "./components/ContactsAdd";
+import ContactsView from "./components/ContactsView";
+import "./styles/styles.css";
 
 export default function App() {
-  const [contacts, setContacts] = useState([])
-  
-  //TODO: Load all contacts on useEffect when component first renders
+  const [contacts, setContacts] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    fetch("http://localhost:4000/contacts")
+      .then((res) => res.json())
+      .then((json) => {
+        setContacts(json);
+        setLoading(false);
+      });
+  }, []);
```

Comment:
I'm curious if this approach works, can you see the loading spinner? I tried different approaches to make this but yours looks much simpler and easier to understand.

Reply:
It looks interesting, the loading stops as soon the content is fetched and then rendered

Reply:
I couldn't test, the app crashes because it can't resolve the spinner. Might not be included in the dependencies or something.